source/develop/release-management/features/virt/nvdimm-device.md

```diff
@@ -0,0 +1,67 @@
+---
+title: NVDIMM devices
+category: feature
+authors: mzamazal
+feature_name: NVDIMM devices
+feature_modules: engine,vdsm
+feature_status: In development
+---
+
+# NVDIMM devices
+
+## Summary
+
+This feature allows adding NVDIMM (non-volatile memory) devices to VMs.
+
+## Owner
+
+*   Name: Milan Zamazal
+*   Email: mzamazal@redhat.com
+
+## Description
+
+NVDIMM devices provide persistent RAM.  QEMU and libvirt allow to add an emulated NVDIMM device to a VM, backed by a real NVDIMM device on the host to ensure data persistence.
+
+This feature allows assigning host NVDIMM devices to VMs.  A new NVDIMM category of host devices is added (although technically NVDIMM devices are treated as memory rather than host devices by libvirt).
+
+The NVDIMM device in the guest is emulated so its mode, partitioning, etc. can be reconfigured in the guest independently of the host device settings.
+
+Information about the emulation details can be found in
+[QEMU NVDIMM documentation](https://github.com/qemu/qemu/blob/master/docs/nvdimm.txt).
+
+## Prerequisites
+
+NVDIMM hardware is needed to test this feature.  To a limited extent, QEMU emulated devices in VMs used as hosts can be used for testing during implementation.
+
+## Limitations
+
+Memory snapshots are disabled when an NVDIMM device is present in a VM.  There is no way to make a snapshot of NVDIMM content and a memory snapshot cannot work correctly without having the corresponding NVDIMM data.  If a memory snapshot is attempted, a non-memory snapshot is created instead.
```

Comment:
iirc, creating a memory snapshot for a VM with NVDIMM device would fail with a validation error, no?

Reply:
No. See https://gerrit.ovirt.org/c/109057/8/frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/dataprovider/AsyncDataProvider.java#525, I don't know whether it's easily fixable but if it is I need some help with that.

Reply:
ok, sorry - I missed those comments somehow
commented on the patch now - if we move the validation from isMemorySnapshotSupported to that place in the validate method, we'll return an error and be consistent with other flows where memory snapshot cannot be taken due to the way the VM is configured

Reply:
Fixed in the posted patch, the corresponding comment removed from the feature page.